projects/plugins/jetpack/modules/related-posts/related-posts.js

```diff
@@ -276,52 +286,75 @@
 	function startRelatedPosts() {
 		jprp.cleanupTrackedUrl();
 
-		var endpointURL = jprp.getEndpointURL(),
-			$relatedPosts = $( '#jp-relatedposts' );
+		var endpointURL = jprp.getEndpointURL();
+		var relatedPosts = document.querySelector( '#jp-relatedposts' );
 
-		if ( $( '#jp-relatedposts .jp-relatedposts-post' ).length ) {
+		if ( document.querySelectorAll( '#jp-relatedposts .jp-relatedposts-post' ).length ) {
 			afterPostsHaveLoaded();
 			return;
 		}
 
-		$.getJSON( endpointURL, function ( response ) {
-			if ( 0 === response.items.length || 0 === $relatedPosts.length ) {
-				return;
-			}
+		var request = new XMLHttpRequest();
+		request.open( 'GET', endpointURL, true );
+		request.setRequestHeader( 'x-requested-with', 'XMLHttpRequest' );
+
+		request.onreadystatechange = function () {
```

Comment:
It definitely would have been prettier, yes. Particularly because of the built-in support for `.json()`.

However, not only is `fetch` unsupported in IE 11, it's a promise-based API, and promises are also unsupported in IE 11. This means we'd need polyfills for:
- `fetch` itself
- Promises
- Generators (at least for Babel's implementation of promises, which is based on generators)

That's starting to look like a lot of code for a single AJAX request 😕 

I plan to go over these files again whenever Core decides to drop IE 11 support (which in my opinion should have already happened), and reimplement them using ES 2015+. Replacing XHR with `fetch` will definitely be a part of that 👍

Reply:
I thought so 😉 Thanks for confirming.

Code does look great in my opinion. Once you get someone to test it, feel free to 🚢 . Lovely work!